File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
       set<char>st;
       int maxlength=0;
       int left=0;
       for(int right=0;right<s.size();right++){
         while(st.find(s[right])!=st.end()){
            st.erase(s[left]);
            left++;
         }
         st.insert(s[right]);
         maxlength=max(maxlength,right-left+1);
       }
       return maxlength; 
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        start=0
        st=set()
        maxans=0
        for x in s:
            maxans=max(maxans,len(st))
            while x in st:
                st.remove(s[start])
                start+=1
            st.add(x)
        maxans=max(maxans,len(st))
        return maxans       